src/Adapter/SymfonyCache.php

```diff
@@ -169,6 +178,28 @@ public function cacheAction(string $token, string $type): Response
 
         $path = 'all' === $type ? $this->cacheDir : sprintf('%s/%s', $this->cacheDir, $type);
 
+        /*
+         * NEXT_MAJOR: Remove this if condition and else block.
+         * clean up event dispatcher just before clearing the cache to prevent further events execution
+         */
+        if (null !== $this->eventDispatcher) {
+            foreach ($this->eventDispatcher->getListeners() as $eventName => $eventListenerList) {
+                foreach ($eventListenerList as $listener) {
+                    $listener[0] instanceof EventSubscriberInterface ?
+                        $this->eventDispatcher->removeSubscriber($listener[0]) :
+                        $this->eventDispatcher->removeListener($eventName, [$listener[0], $listener[1]]);
+                }
+            }
+        } else {
+            @trigger_error(sprintf(<<<'MESSAGE'
+Passing no 9th argument to %s is deprecated since version 3.1 and will be mandatory in 4.0.
```

Comment:
Let's add the package name, and I will add the version number when releasing.

```suggestion
Passing no 9th argument to %s is deprecated since version sonata-project/cache-bundle 3.x and will be mandatory in 4.0.
```